internal/server/server.go

```diff
@@ -13,161 +13,116 @@ import (
 	"github.com/osbuild/image-builder/internal/cloudapi"
 
 	"github.com/getkin/kin-openapi/openapi3"
-	"github.com/go-chi/chi"
+	"github.com/labstack/echo/v4"
 )
 
 type Handlers struct{}
 
-func (s *Handlers) GetVersion(w http.ResponseWriter, r *http.Request) {
+func (s *Handlers) GetVersion(ctx echo.Context) error {
 	spec, err := GetSwagger()
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
```

Comment:
Just returning `err` is enough here. Echo turns non`HTTPError`-errors into internal server errors automatically.

Reply:
neat!